### ISSUE TITLE
Typescript: wrong return type getDateDiff

### ISSUE THREAD

**Issue by mesqueeb** [quasarframework/quasar#6246]
getDateDiff says it returns a `Date` while it actually returns a `number`

**Comment by IlCallo** [comment]
Fix is in https://github.com/quasarframework/quasar/pull/5729, we must wait @rstoenescu to check what I wrote there and provide feedback, because some implementation updates are needed too.

The one you requested is a little update which doesn't involve a breaking change (all others in that PR are breaking changes, that's why it's still there stuck), so I guess I can put it into another PR

**Comment by rstoenescu** [comment]
Merged PR (thanks for it!) and it will be available in upcoming "quasar" v1.8.4.

**Comment by mesqueeb** [comment]
@rstoenescu that PR didnt have the fix. It still says:
```
function getDateDiff(date: Date | number | string, subtract: Date | number | string, unit?: DateUnitOptions): Date;
```

**Comment by rstoenescu** [comment]
Closing with new PR.
Will be available in "quasar" v1.8.4.
